### PR TITLE
add dpcpp toolchain

### PR DESCRIPTION
**Pull request by jonathanpoelen**

### CODE REVIEW

File: xmake/toolchains/dpcpp/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+-- @author      ruki
+-- @file        xmake.lua
+--
+
+-- define toolchain
+toolchain("dpcpp")
+
+    -- set homepage
+    set_homepage("https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler.html")
+    set_description("Intel LLVM C/C++ Compiler")
```
Comment (waruqi) [inline xmake/toolchains/dpcpp/xmake.lua:26]: It is also the `Intel LLVM C/C++ Compiler` ? What is the difference between it and icx?

Comment (waruqi) [reply xmake/toolchains/dpcpp/xmake.lua:26]: Could you use a different description to distinguish between icx and dpcc?

Comment (jonathanpoelen) [reply xmake/toolchains/dpcpp/xmake.lua]: yes, it's another intel compiler that uses SYCL.

But I just realized that it only allows to compile C sources. Is removing `toolchain:set("toolset", "cc", "dpcpp")` enough ? Should I remove `set_kind("standalone")` ?

Comment (waruqi) [reply xmake/toolchains/dpcpp/xmake.lua]: need not remove it

**Pull request merged**